src/lando/utils/landing_checks.py

```diff
@@ -0,0 +1,499 @@
+
+        if status_code == 401:
+            return (
+                f"Your commit message references bug {bug_id}, which is currently private. To avoid "
+                "disclosing the nature of this bug publicly, please remove the affected bug ID "
+                f"from the commit message. {BMO_SKIP_HINT}"
+            )
+
+        if status_code == 404:
+            return (
+                f"Your commit message references bug {bug_id}, which does not exist. "
+                f"Please check your commit message and try again. {BMO_SKIP_HINT}"
+            )
+
+        return (
+            f"While checking if bug {bug_id} in your commit message is a security bug, "
+            f"an error occurred and the bug could not be verified. {BMO_SKIP_HINT}"
+        )
+
+
+@dataclass
+class PatchCollectionAssessor:
+    """Assess pushes for landing issues."""
+
+    patch_helpers: Iterable[PatchHelper]
+    push_user_email: str | None = None
+
+    def run_patch_collection_checks(
+        self,
+        patch_collection_checks: list[type[PatchCollectionCheck]],
+        patch_checks: list[type[PatchCheck]],
+    ) -> list[str]:
+        """Execute the set of checks on the diffs, returning a list of issues.
+
+        `patch_collection_checks` specifies the collection-wide checks to run on the
+        `patch_helpers`. `patch_checks` specifies the checks to run on individual
+        patches.
+        """
+        issues = []
+
+        checks = [check(self.push_user_email) for check in patch_collection_checks]
+
+        for patch_helper in self.patch_helpers:
+            # Pass the patch information into the push-wide check.
+            for check in checks:
+                check.next_diff(patch_helper)
+
+            parsed_diff = rs_parsepatch.get_diffs(patch_helper.get_diff())
+
+            author, email = patch_helper.parse_author_information()
+
+            # Run diff-wide checks.
+            diff_assessor = DiffAssessor(
+                author=author,
+                email=email,
+                commit_message=patch_helper.get_commit_description(),
+                parsed_diff=parsed_diff,
+            )
+            if diff_issues := diff_assessor.run_diff_checks(patch_checks):
+                issues.extend(diff_issues)
+
+        # Collect the result of the push-wide checks.
+        for check in checks:
+            if issue := check.result():
+                issues.append(issue)
+
+        return issues
+
+
+ALL_STACK_CHECKS = PatchCollectionCheck.__subclasses__()
+ALL_COMMIT_CHECKS = PatchCheck.__subclasses__()
+ALL_CHECKS = ALL_COMMIT_CHECKS + ALL_STACK_CHECKS
+
+
+class LandingChecks:
+    """Utility class to run landing checks (a.k.a. hooks) on a list of commits."""
+
+    requester_email: str
+
+    def __init__(self, requester_email: str):  # noqa: ANN001
+        self.requester_email = requester_email
+
+    def run(
+        self,
+        hooks: list[str] | list[type[PatchCheck] | type[PatchCollectionCheck]],
```

Comment:
We accept either a string hook name or the concrete object when running hooks here, but we only ever call this method with the concrete objects in a test. I think we should remove the ability to pass `list[type[PatchCheck] | type[PatchCollectionCheck]]` and only accept `list[str]` to keep things simple.